src/utils/ddb/ddb_mgmt.c

```diff
@@ -0,0 +1,247 @@
+	uint64_t              pool_size;
+	uint64_t              total_size;
+	const unsigned long   GiB = (1ul << 30);
+
+	D_ASSERT(scm_mount_size != NULL);
+	D_ASSERT(bio_nvme_configured(SMD_DEV_TYPE_META));
+	D_INIT_LIST_HEAD(&pool_list);
+	rc = smd_pool_list(&pool_list, &pool_list_cnt);
+	if (rc != 0) {
+		D_ERROR("Failed to get pool info list from SMD\n");
+		return rc;
+	}
+
+	total_size = 0;
+	d_list_for_each_entry_safe(pool_info, tmp, &pool_list, spi_link) {
+		if ((pool_info->spi_blob_sz[SMD_DEV_TYPE_META] == 0) ||
+		    (pool_info->spi_flags[SMD_DEV_TYPE_META] & SMD_POOL_IN_CREATION)) {
+			continue;
+		}
+		D_ASSERT(pool_info->spi_scm_sz > 0);
+
+		/** Align to 4K */
+		pool_size = (D_ALIGNUP(pool_info->spi_scm_sz, 1ULL << 12)) *
+			    pool_info->spi_tgt_cnt[SMD_DEV_TYPE_META];
+		total_size += pool_size;
+		D_INFO("Pool " DF_UUID " required scm size: " DF_U64 "", DP_UUID(pool_info->spi_id),
+		       pool_size);
+	}
+
+	d_list_for_each_entry_safe(pool_info, tmp, &pool_list, spi_link) {
+		d_list_del(&pool_info->spi_link);
+		/* Frees spi_tgts, spi_blobs, and pool_info */
+		smd_pool_free_info(pool_info);
+	}
+
+	*scm_mount_size = (D_ALIGNUP(total_size, GiB) / GiB);
+	return rc;
+}
+
+static int
+destroy_cb(const char *path, const struct stat *sb, int flag, struct FTW *ftwbuf)
+{
+	int rc;
+
+	if (ftwbuf->level == 0)
+		return 0;
+
+	switch (flag) {
+	case FTW_DP:
+		rc = rmdir(path);
+		break;
+	case FTW_F:
+		rc = unlink(path);
+		break;
+	default:
+		D_ERROR("Cannot delete %s. Unexpected file type: %#x", path, flag);
+		return EINVAL;
+	}
+
+	if (rc)
+		D_ERROR("Failed to remove %s", path);
+	return rc;
+}
+
+int
+ddb_clear_dir(const char *dir)
+{
+	int rc;
+
+	rc = nftw(dir, destroy_cb, 32, FTW_DEPTH | FTW_PHYS | FTW_MOUNT);
+	if (rc)
+		rc = (errno == 0 ? rc : daos_errno2der(errno));
+
+	return rc;
+}
+
+int
+ddb_is_mountpoint(const char *path)
+{
+	int           rc = 0;
+	struct stat   st_path, st_parent;
+	char          parent_path[DDB_PROV_MEM_BUF_MAX];
+
+	D_ASSERT(path != NULL);
+	if (access(path, F_OK) != 0)
+		return daos_errno2der(errno);
+
+	rc = snprintf(parent_path, sizeof(parent_path), "%s/..", path);
+	if (unlikely(rc >= sizeof(parent_path)))
+		return -DER_EXCEEDS_PATH_LEN;
+
+	rc = stat(path, &st_path);
+	if (rc)
+		return daos_errno2der(errno);
+
+	rc = stat(parent_path, &st_parent);
+	if (rc)
+		return daos_errno2der(errno);
+
+	return (st_path.st_dev == st_parent.st_dev) ? 0 : 1;
+}
+
+int
+ddb_recreate_pooltgts(const char *scm_mount)
+{
+	struct smd_pool_info *pool_info = NULL;
+	struct smd_pool_info *tmp;
+	d_list_t              pool_list;
+	int                   rc = 0;
+	int                   pool_list_cnt;
+
+	D_ASSERT(bio_nvme_configured(SMD_DEV_TYPE_META));
+	D_INIT_LIST_HEAD(&pool_list);
+	rc = smd_pool_list(&pool_list, &pool_list_cnt);
+	if (rc != 0) {
+		D_ERROR("Failed to get pool info list from SMD");
+		return rc;
+	}
+
+	d_list_for_each_entry_safe(pool_info, tmp, &pool_list, spi_link) {
```

Comment:
You do not call `d_list_del(&pool_info->spi_link);` in this `for` section, then no need `_safe` version.